tests/docker/test_parsing.py

```diff
@@ -0,0 +1,12 @@
+import pytest
+from prefect.docker import parse_image_tag
+from itertools import product
+
+image_names = ['some_url_to_a_docker_repo.com:5050/path/to/image','some_url_to_a_docker_repo.com/path/to/image']
+image_tags = [[], ['tag1']]
+@pytest.mark.parametrize('docker_image,image_name,image_tag', [(':'.join([image]+tag), image, tag) for image, tag in product(image_names, image_tags)])
+def test_docker_image_parsing(docker_image,image_name,image_tag):
```

Comment:
@MSSandroid thanks for adding the tests and moving it to a separate function! I'm reviewing this now, so I may have a few more comments in a bit. 

A couple of things on the tests - a lot of us (and other users) wind up going straight to the tests to quickly determine how a function is supposed to behave, because we can see the expected inputs and outputs. Pytest's parameterize is really helpful for this. It's also helpful because it allows us to at a glance check for edge case coverage. 

The way your have your values setup inside of parameterize, it takes a quite a bit of effort to understand what the input is, and what the output is. Could you explicitly list the input values and the expected outputs similar to 
```
    @pytest.mark.parametrize(
        "value,expected",
        [
            (<input_value_1>, <expected_output_1>),
            (<input_value_2>, <expected_output_2>),
            (<input_value_3>, <expected_output_3>),
        ],
    )
    def test_my_func(self, value, expected)
```

Could you also add a couple of parameterized tests to make sure that your error is being raised under the expected conditions?